library/helpers/isHTTPAuthScheme.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ *  https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml
+ */
+export type HTTPAuthScheme =
+  | "basic"
+  | "bearer"
+  | "digest"
+  | "dpop"
+  | "gnap"
+  | "hoba"
+  | "mutal"
+  | "negotiate"
+  | "privatetoken"
+  | "scram-sha-1"
+  | "scram-sha-256"
+  | "vapid";
```

Comment:
I like this technique to avoid types and data to go out of sync: https://hansott.codes/blog/how-to-check-if-string-is-member-of-union-type-19

Reply:
Cool to have your own blog article to link to :)